SU2_CFD/src/output/COutput.cpp

```diff
@@ -679,6 +680,39 @@ void COutput::WriteToFile(CConfig *config, CGeometry *geometry, unsigned short f
 
       break;
 
+    case CGNS:
+
+      if (fileName.empty()) fileName = config->GetFilename(volumeFilename, "", curTimeIter);
+
+      /*--- Load and sort the output data and connectivity. ---*/
+      volumeDataSorter->SortConnectivity(config, geometry, true);
+
+      /*--- Write CGNS ---*/
+      if (rank == MASTER_NODE) {
+        (*fileWritingTable) << "CGNS" << fileName + CCGNSFileWriter::fileExt;
+      }
+
+      fileWriter = new CCGNSFileWriter(fileName, volumeDataSorter);
```

Comment:
Throw an error in the constructor if SU2 was compiled without cgns support, or maybe earlier in the code would be better (not sure where but have a look at what we do for tecplot).